Modules/Gyro/gyro_factory.py

```diff
@@ -0,0 +1,14 @@
+from . import config
+
+if config.GYRO_CONNECTED:
+    from . import mpu9250
+else:
+    from . import fake_gyro
+
+#this function returns the mpu9250 hardware module object if the hardware is connected.
+#otherwise returns and object that simulates hardware 
+def initialize():
+    if(config.GYRO_CONNECTED == True):
```

Comment:
Same comment as up there. The `GYRO_CONNECTED` check should be executed once as a preparation check and not in run-time.